test/plugin/base_test.rb

```diff
@@ -626,6 +654,118 @@ def test_struct_payload_json_container_log
     end
   end
 
+  # Verify that we drop the log entries when 'require_valid_tags' is true and
+  # any non-string tags or tags with non-utf8 characters are detected.
+  def test_reject_invalid_tags_with_require_valid_tags_true
+    setup_gce_metadata_stubs
+    invalid_tags = NON_STRING_TAGS.merge(EMPTY_TAG).merge(NON_UTF8_TAGS)
+    invalid_tags.each do |tag, _|
+      setup_logging_stubs do
+        @logs_sent = []
+        d = create_driver(REQUIRE_VALID_TAGS_CONFIG, tag)
+        d.emit('msg' => log_entry(0))
+        d.run
+      end
+      verify_log_entries(0, COMPUTE_PARAMS, 'structPayload')
+    end
+  end
+
+  # Verify that tags are properly encoded. When 'require_valid_tags' is true, we
+  # only accept string tags with utf8 characters.
+  def test_encode_tags_with_require_valid_tags_true
+    setup_gce_metadata_stubs
+    VALID_TAGS.each do |tag, encoded_tag|
+      setup_logging_stubs([COMPUTE_PARAMS.merge(log_name: encoded_tag)]) do
+        @logs_sent = []
+        d = create_driver(REQUIRE_VALID_TAGS_CONFIG, tag)
+        d.emit('msg' => log_entry(0))
+        d.run
+      end
+      verify_log_entries(1, COMPUTE_PARAMS, 'structPayload')
+      # Each batch in the logs sent shares the same log name, so we only verify
+      # the first one.
+      assert_equal "projects/#{PROJECT_ID}/logs/#{encoded_tag}",
+                   @logs_sent[0]['logName']
+    end
+  end
+
+  # Verify that tags extracted from container names are properly encoded.
+  def test_encode_tags_from_container_name_with_require_valid_tags_true
```

Comment:
Yeah, it's possible. but the container name is a part of the tag, and the tag got rejected (if `require_valid_tags` is true) or converted to non-utf-8 (if `require_valid_tags` is false) before reaching the kubernete specific logic (use the regex to extract the container name).

One of the issues in the bug ticket is that the kubernetes regex `match` function would throw exceptions for non-utf8 characters.

Do you think we should test that (container name with non-utf8 characters) anyway just to make sure it's behaving as expected?